pkg/auth/authz_handler.go

```diff
@@ -65,18 +87,35 @@ func (a *AuthzHandler) Authorizer(request *http.Request) (err error) {
 		isAuthorized = payload.(*ocm.AuthPayload).IsAuthorized
 	} else {
 		if isAuthorized, err = a.allowedToUseAssistedInstaller(username); err != nil {
-			return common.NewInfraError(http.StatusInternalServerError, err)
+			return common.NewApiError(http.StatusInternalServerError, err)
 		}
 	}
 
 	if !isAuthorized {
 		return common.NewInfraError(
-			http.StatusUnauthorized,
+			http.StatusForbidden,
 			fmt.Errorf(
 				"%s: Unauthorized to access route (access review failed)",
 				username))
 	}
-	return
+
+	if payload.Role == ocm.UserRole {
```

Comment:
Maybe we could add some unit-tests for this logic now that it's AuthzHandler? Or the subsystem tests should already cover it actually. Any way, can tackle later on.